File: backend/api/schemas/insights.py
"""
Insights API Schemas

SPRINT3-T3.1: Schemas for InsightsEngine and /insights/* endpoints
"""
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class HotLeadAction(BaseModel):
    """Lead ready to close with actionable context"""
    follower_id: str
    name: str
    username: str
    profile_pic_url: Optional[str] = None
    last_message: str
    hours_ago: int
    product: Optional[str] = None
    deal_value: float = 0.0
    context: str  # "Madre de 3, ya aceptó cuotas"
    action: str   # "Envíale el link de pago"
    purchase_intent_score: float = 0.0


class BookingInfo(BaseModel):
    """Today's booking info"""
    id: str
    title: str
    time: str
    attendee_name: str
    attendee_email: Optional[str] = None
    platform: str


class TodayMission(BaseModel):
    """Daily mission with actionable priorities"""
    potential_revenue: float
    hot_leads: List[HotLeadAction]
    pending_responses: int
    today_bookings: List[BookingInfo]
    ghost_reactivation_count: int = 0


class ContentInsight(BaseModel):
    """Most asked topic insight"""
    topic: str
    count: int
    percentage: float
    quotes: List[str]
    suggestion: str


class TrendInsight(BaseModel):
    """Emerging trend insight"""
    term: str
    count: int
    growth: str  # "+50%" or "new"
    suggestion: str


class ProductInsight(BaseModel):
    """Product demand insight"""
    product_name: str
    count: int
    potential_revenue: float
    suggestion: str


class CompetitionInsight(BaseModel):
    """Competitor mentions insight"""
    competitor: str
    count: int
    sentiment: str  # "positive", "neutral", "negative"
    suggestion: str


class WeeklyInsights(BaseModel):
    """Four insight cards for weekly analysis"""
    content: Optional[ContentInsight] = None
    trend: Optional[TrendInsight] = None
    product: Optional[ProductInsight] = None
    competition: Optional[CompetitionInsight] = None


class WeeklyMetrics(BaseModel):
    """Weekly metrics with deltas vs previous week"""
    revenue: float = 0.0
    revenue_delta: float = 0.0  # percentage change
    sales_count: int = 0
    sales_delta: int = 0  # absolute change
    response_rate: float = 0.0  # 0-1
    response_delta: float = 0.0  # percentage points
    hot_leads_count: int = 0
    conversations_count: int = 0
    new_leads_count: int = 0

File: frontend/src/services/api/insights.ts
import { apiFetch } from "./client";

export interface HotLeadAction {
  follower_id: string;
  name: string;
  username: string;
  profile_pic_url?: string;
  last_message: string;
  hours_ago: number;
  product?: string;
  deal_value: number;
  context: string;
  action: string;
  purchase_intent_score: number;
}

export interface BookingInfo {
  id: string;
  title: string;
  time: string;
  attendee_name: string;
  attendee_email?: string;
  platform: string;
}

export interface TodayMission {
  potential_revenue: number;
  hot_leads: HotLeadAction[];
  pending_responses: number;
  today_bookings: BookingInfo[];
  ghost_reactivation_count: number;
}

export interface ContentInsight {
  topic: string;
  count: number;
  percentage: number;
  quotes: string[];
  suggestion: string;
}

export interface TrendInsight {
  term: string;
  count: number;
  growth: string;
  suggestion: string;
}

export interface ProductInsight {
  product_name: string;
  count: number;
  potential_revenue: number;
  suggestion: string;
}

export interface CompetitionInsight {
  competitor: string;
  count: number;
  sentiment: string;
  suggestion: string;
}

export interface WeeklyInsights {
  content?: ContentInsight;
  trend?: TrendInsight;
  product?: ProductInsight;
  competition?: CompetitionInsight;
}

export interface WeeklyMetrics {
  revenue: number;
  revenue_delta: number;
  sales_count: number;
  sales_delta: number;
  response_rate: number;
  response_delta: number;
  hot_leads_count: number;
  conversations_count: number;
  new_leads_count: number;
}

export async function getTodayMission(creatorId: string): Promise<TodayMission> {
  return apiFetch(`/insights/${creatorId}/today`);
}

export async function getWeeklyInsights(creatorId: string): Promise<WeeklyInsights> {
  return apiFetch(`/insights/${creatorId}/weekly`);
}

export async function getWeeklyMetrics(creatorId: string): Promise<WeeklyMetrics> {
  return apiFetch(`/insights/${creatorId}/metrics`);
}
